### PR TITLE
refactor/optimize feed template/css, add color to brand landing page header (bug 1131721)

### PR DESCRIPTION
**Pull request by ngokevin**
Merges brand landing into collection landing. Now shelves, brands, and collections all share the same template and CSS.

- get rid of stuff fireplace doesn't need in feed.js
- add a feed item transformer to handle more presentation logic and offload stuff from templates
- lower test timeout to 5s
- remove ```feed_``` prefix from templates/views to match the path
- remove unused stuff from feed landing views
- move desktop promo stuff to separate module
- remove all collection-landing specific classes
- optimize feed colors CSS by a huge amount

screen shot 2015-02-13 at 4 27 30 pm
screen shot 2015-02-13 at 4 28 10 pm

### CODE REVIEW

File: src/media/js/feed.js [hunk 1]
```diff
@@ -4,42 +4,74 @@
...
+                        break;
+                    case 'preview':
+                        feedItem.isPreview = true;
+                        break;
+                }
+                feedItem.src = 'featured-app';
+                break;
+            case 'brand':
+                feedItem.color = get_brand_color_class(feedItem);
+                feedItem.name = brands.get_brand_type(feedItem.type, feedItem.apps.length);
```
Comment (chuckharmston) [inline src/media/js/feed.js]: So is @houndci actually the robotic manifestation of @cvan?

Comment (ngokevin) [inline src/media/js/feed.js]: i knew this felt nostalgic :)

**Comment by ngokevin** [comment]
@spasovski r?

File: src/media/js/feed.js (hunk 1 above)
Comment (cvan) [inline src/media/js/feed.js]: god i love it

File: src/media/css/feed/brands.styl [hunk 2]
```diff
@@ -40,6 +40,10 @@
     }
 }
 
+.brand-landing-icon {
+    background-color: rgba($greyscale-light-grey);
```
Comment (spasovski) [inline src/media/css/feed/brands.styl]: just `$greyscale-light-grey` is fine

File: src/media/css/feed/index.styl [hunk 3]
```diff
@@ -2,19 +2,7 @@ collection_background_colors($color) {
     .feed-landing-header {
         background-color: rgba($color, 0.55);
     }
-    &[data-collection-landing-is-listing] {
```
Comment (spasovski) [inline src/media/css/feed/index.styl]: Nice. These deletions probably shrunk the CSS size a bit

File: src/media/js/desktop_promo.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+                name: 'productivity',
+                url: urls.reverse('feed/collection', ['productivity-appsdesktop']),
+                text: gettext('Productivity Apps—Desktop Essentials'),
+            },
+            {
+                name: 'box',
+                url: urls.reverse('app', ['box']),
+                text: gettext('Store and share any type of file with Box.'),
+            },
+        ].map(function (item) {
```
Comment (spasovski) [inline src/media/js/desktop_promo.js]: `function(item) {`

File: src/media/js/views/feed/editorial.js [hunk 5]
```diff
@@ -1,23 +1,23 @@
-define('views/feed/feed_brand',
-    ['feed', 'jquery', 'l10n', 'utils', 'z'],
-    function(feed, $, l10n, utils, z) {
+define('views/feed/editorial',
```
Comment (spasovski) [inline src/media/js/views/feed/editorial.js]: liking the renaming

File: src/templates/_macros/feed_item.html [hunk 6]
```diff
@@ -145,20 +106,22 @@ <h1 class="name">{{ feed_app.app.name|translate(feed_app) }}</h1>
...
       </ul>
     {% endif %}
-    <div class="curve" style="background-color: {{ inline_color|hex2rgba('.55') }}"></div>
-    <div class="info" style="background-color: {{ inline_color|hex2rgba('.55') }}">
+    <div class="feed-curve curve" style="background-color: {{ coll.inline_color|hex2rgba('.55') }}"></div>
+    <div class="info" style="background-color: {{ coll.inline_color|hex2rgba('.55') }}">
       <h1 class="name">{{ coll.name|translate(coll)|safe }}</h1>
     </div>
-    <div class="alt-curve" style="background-color: {{ inline_color }}"></div>
+    <div class="alt-curve" style="background-color: {{ coll.inline_color }}"></div>
```
Comment (spasovski) [inline src/templates/_macros/feed_item.html]: You'll hit some collisions here with my commit that already landed. Should be easy though.

**Comment by spasovski** [comment]
I see this



and this 



But the fixes to both of these should be pretty trivial. I would say we can probably merge anyway and we can address those two issues (one of which fails in Travis anyway)

**Comment by ngokevin** [comment]
ah. i can fix those before merge, there's no rush atm

**Pull request merged**